src/components/viewmodels/rooms/RoomSummaryCardViewModel.tsx

```diff
@@ -0,0 +1,290 @@
+import { ReportRoomDialog } from "../../views/dialogs/ReportRoomDialog";
+import { Key } from "../../../Keyboard";
+import { onRoomTopicLinkClick } from "../../views/elements/RoomTopic";
+import { usePinnedEvents } from "../../../hooks/usePinnedEvents";
+import { tagRoom } from "../../../utils/room/tagRoom";
+import { inviteToRoom } from "../../../utils/room/inviteToRoom";
+
+export interface RoomSummaryCardState {
+    isDirectMessage: boolean;
+    isRoomEncrypted: boolean;
+    e2eStatus: E2EStatus | undefined;
+    isVideoRoom: boolean;
+    roomJoinRule: JoinRule;
+    alias: string;
+    isFavorite: boolean;
+    canInviteToState: boolean;
+    pinCount: number;
+    searchInputRef: React.RefObject<HTMLInputElement | null>;
+    onUpdateSearchInput: (e: React.KeyboardEvent<HTMLInputElement>) => void;
+    onRoomMembersClick: () => void;
+    onRoomThreadsClick: () => void;
+    onRoomFilesClick: () => void;
+    onRoomExtensionsClick: () => void;
+    onRoomPinsClick: () => void;
+    onRoomSettingsClick: (ev: Event) => void;
+    onLeaveRoomClick: () => void;
+    onShareRoomClick: () => void;
+    onRoomExportClick: () => Promise<void>;
+    onRoomPollHistoryClick: () => void;
+    onReportRoomClick: () => Promise<void>;
+    onFavoriteToggleClick: () => void;
+    onInviteToRoomClick: () => void;
+}
+
+export function useRoomSummaryCardViewModel(
+    room: Room,
+    permalinkCreator: RoomPermalinkCreator,
+    onSearchCancel?: () => void,
+): RoomSummaryCardState {
+    const cli = useMatrixClientContext();
+
+    const isRoomEncrypted = useIsEncrypted(cli, room) ?? false;
+    const roomContext = useScopedRoomContext("e2eStatus", "timelineRenderingType");
+    const e2eStatus = roomContext.e2eStatus;
+    const isVideoRoom = calcIsVideoRoom(room);
+
+    const roomState = useRoomState(room);
+    // used to check if the room is public or not
+    const roomJoinRule = roomState.getJoinRule();
+    const alias = room.getCanonicalAlias() || room.getAltAliases()[0] || "";
+    const pinCount = usePinnedEvents(room).length;
+    // value to check if the user can invite to the room
+    const canInviteToState = useEventEmitterState(room, RoomStateEvent.Update, () => canInviteTo(room));
+
+    const roomTags = useEventEmitterState(RoomListStore.instance, LISTS_UPDATE_EVENT, () =>
+        RoomListStore.instance.getTagsForRoom(room),
+    );
+    const isFavorite = roomTags.includes(DefaultTagID.Favourite);
+
+    const directRoomsList = useAccountData<Record<string, string[]>>(room.client, EventType.Direct);
+    const [isDirectMessage, setDirectMessage] = useState(false);
+    // Check if room is DM, and set the room as a direct message if it is
+    useEffect(() => {
+        for (const [, dmRoomList] of Object.entries(directRoomsList)) {
+            if (dmRoomList.includes(room?.roomId ?? "")) {
+                setDirectMessage(true);
+                break;
+            }
+        }
+    }, [room, directRoomsList]);
```

Comment:
It could be great to move it to its own custom hook in this file. In order to help the readability of this hook.